src/packagedcode/debian.py

```diff
@@ -402,10 +402,11 @@ def assemble(cls, package_data, resource, codebase):
                 continue
 
             for pkgdt in res.package_data:
+                pkgdt = models.PackageData.from_dict(pkgdt)
```

Comment:
I wonder if this could be done directly in package.update where we could accept `package_data` as a mapping or as an object and convert to object if needed?

Reply:
Good point, I've updated `Package.update()` to convert the package data to a `PackageData` object if the incoming package data is in a dictionary.